lib/AST/ASTPrinter.cpp

```diff
@@ -4043,15 +4069,472 @@ void PrintAST::visitMissingMemberDecl(MissingMemberDecl *decl) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitLinearFunctionExpr(LinearFunctionExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitDefaultArgumentExpr(DefaultArgumentExpr *expr) {
+}
+
+void PrintAST::visitLazyInitializerExpr(LazyInitializerExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitOpenExistentialExpr(OpenExistentialExpr *expr) {
+  visit(expr->getValueProvidingExpr());
+  visit(expr->getExistentialValue());
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitStringToPointerExpr(StringToPointerExpr *expr) {
+}
+
+void PrintAST::visitVarargExpansionExpr(VarargExpansionExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitArchetypeToSuperExpr(ArchetypeToSuperExpr *expr) {
+}
+
+void PrintAST::visitDestructureTupleExpr(DestructureTupleExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitDynamicMemberRefExpr(DynamicMemberRefExpr *expr) {
+}
+
+void PrintAST::visitDynamicSubscriptExpr(DynamicSubscriptExpr *expr) {
+}
+
+void PrintAST::visitPointerToPointerExpr(PointerToPointerExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitUnresolvedMemberExpr(UnresolvedMemberExpr *expr) {
+}
+
+void PrintAST::visitDiscardAssignmentExpr(DiscardAssignmentExpr *expr) {
+}
+
+void PrintAST::visitEditorPlaceholderExpr(EditorPlaceholderExpr *expr) {
+}
+
+void PrintAST::visitForcedCheckedCastExpr(ForcedCheckedCastExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitOverloadedDeclRefExpr(OverloadedDeclRefExpr *expr) {
+}
+
+void PrintAST::visitUnresolvedDeclRefExpr(UnresolvedDeclRefExpr *expr) {
+}
+
+void PrintAST::visitUnresolvedPatternExpr(UnresolvedPatternExpr *expr) {
+}
+
+void PrintAST::visitAnyHashableErasureExpr(AnyHashableErasureExpr *expr) {
+}
+
+void PrintAST::visitConstructorRefCallExpr(ConstructorRefCallExpr *expr) {
+  if (auto type = expr->getType()) {
+    if (auto *funcType = type->getAs<FunctionType>()) {
+      printType(funcType->getResult());
+    }
+  }
+}
+
+void PrintAST::visitFunctionConversionExpr(FunctionConversionExpr *expr) {
+}
+
+void PrintAST::visitInjectIntoOptionalExpr(InjectIntoOptionalExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitKeyPathApplicationExpr(KeyPathApplicationExpr *expr) {
+}
+
+void PrintAST::visitMetatypeConversionExpr(MetatypeConversionExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitOptionalEvaluationExpr(OptionalEvaluationExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitUnderlyingToOpaqueExpr(UnderlyingToOpaqueExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitUnevaluatedInstanceExpr(UnevaluatedInstanceExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitDotSyntaxBaseIgnoredExpr(DotSyntaxBaseIgnoredExpr *expr) {
+}
+
+void PrintAST::visitUnresolvedSpecializeExpr(UnresolvedSpecializeExpr *expr) {
+  visit(expr->getSubExpr());
+}
```

Comment:
Should we print the generic arguments here?

Reply:
I couldn't find an example where this expression is created. 

Generally, I'd like to merge this even if some expressions aren't printed 100% correct yet (there certainly are more parts missing).

Reply:
It's created by the parser for e.g `Array<Int>`, before the types have been resolved. It gets rewritten by the type-checker though, so I don't believe the current frontend printing option will come across it. But if we ever wanted to print the AST prior to type-checking, I think we should print the generic arguments. But yeah, that doesn't need to hold up this PR.